src/TMap.h

```diff
@@ -128,32 +137,50 @@ class TMap : public QObject
     void setMmpMapLocation(const QString &location);
     QString getMmpMapLocation() const;
 
+    // has setRoomNamesShown ever been called on this map?
+    bool getRoomNamesPresent();
+    // show room labels on the map?
+    bool getRoomNamesShown();
+    void setRoomNamesShown(bool shown);
+
+    std::pair<bool, QString> writeJsonMapFile(const QString&);
+    std::pair<bool, QString> readJsonMapFile(const QString&);
+    int getCurrentProgressRoomCount() const { return mProgressDialogRoomsCount; }
+    bool incrementJsonProgressDialog(const bool isExportNotImport, const bool isRoomNotLabel, const int increment = 1);
```

Comment:
![image](https://user-images.githubusercontent.com/110988/107112775-74bfbc00-685a-11eb-9761-251933aae498.png)

This hurts my brain... how about just a simple struct instead?

Reply:
It evolved like that - each argument got added at a separate stage. As it is ***you*** are using some fancy IDE that is documenting the argument names anyhow for these PODs so why complicate things with a `struct`ure which will have to be initialised before being passed anyhow...

Reply:
Sorry, I mean a flag from a struct. This evolved code is hard to read, could be improved.

Otherwise just waiting on the issues reported to be resolved, need any help with them?

Reply:
> Sorry, I mean a flag from a struct. This evolved code is hard to read, could be improved.

I don't get what you mean - could you suggest what you think you want?

:confused: Um, what issues are left that are germane to this PR:
* https://github.com/Mudlet/Mudlet/pull/4546#issuecomment-770277594 is, at best guess, down to a table being stored in lua by code outside of my arena and the order there is random is it not?
* https://github.com/Mudlet/Mudlet/pull/4546#issuecomment-770278308 has been diagnosed and fixed already in #4722 and was the cause of:
* https://github.com/Mudlet/Mudlet/pull/4546#issuecomment-770278483 that is not a feature of this PR and it is being handled elsewhere in #4741.
* https://github.com/Mudlet/Mudlet/pull/4546#issuecomment-770282025 is also not a part of this PR and I have raised an issue #4764 to track it.
* https://github.com/Mudlet/Mudlet/pull/4546#issuecomment-770282292 the first part is also not part of this PR was already known about as #4743 is being resolved by yourself in #4756 - the second part is normal operation as far as I can tell - changing the main background colour does (I think) reset the default background colour for any newly generated text but does not get applied retrospectively to the existing text.